upgrade_test.py

```diff
@@ -236,14 +236,11 @@ def upgrade_node(self, node, upgrade_sstables=True):
                     node.remoter.run(
                         r'sudo apt-get dist-upgrade {} -y '
                         r'-o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" '.format(scylla_pkg))
-        if self.params.get('test_sst3'):
-            node.remoter.run("echo 'enable_sstables_mc_format: true' |sudo tee --append /etc/scylla/scylla.yaml")
-        if self.params.get('test_upgrade_from_installed_3_1_0'):
-            node.remoter.run("echo 'enable_3_1_0_compatibility_mode: true' |sudo tee --append /etc/scylla/scylla.yaml")
         check_reload_systemd_config(node)
         # Current default 300s aren't enough for upgrade test of Debian 9.
         # Related issue: https://github.com/scylladb/scylla-cluster-tests/issues/1726
         node.run_scylla_sysconfig_setup()
+        self._update_scylla_yaml_on_node(node)
```

Comment:
`_update_scylla_yaml_on_node` is not a good name, because it is too general, and doesn't show that it is something very specific...
true that there is another (conditional) change too, but at least we need some documentation explaining what is being updated IMHO

Reply:
also, should you not revert these changes when you rollback?

Reply:
Agreed, it's not the best name. I changed it around again: left the generic name and added the updates dict as a parameter to it. That way we can build the updates dict in the `upgrade_node` method and just pass them to this more generic method.